src/code_translation/NodeSelector.py

```diff
@@ -138,9 +138,10 @@ def resolve_Call(self, node: cst.Call) -> Optional[Node]:
                 result = module.resolve_call(func_name, False, *args, **kwargs)
             elif isinstance(attribute, IRNode):  # calls to IR nodes. e.g. df.join
                 method_name = node.func.attr.value
-                node = attribute
-                module = module_by_name[node.library]
-                result = module.resolve_call(method_name, True, node, *args, **kwargs)
+                if not attribute.library:
+                    raise Exception("Library of IRNode is not set.")
```

Comment:
I think this actually never happens because in resolve_call from inputModule takes care of this